service/server.go

```diff
@@ -187,9 +187,13 @@ func (s *server) ListenAndServe(ctx context.Context) <-chan []error {
 			s.dRunning = false
 			s.mu.Unlock()
 		}()
+	} else {
```

Comment:
I think we should control by wg count instead of using wg.done here. I will push a commit later.